### PR TITLE
Replace JSX with r-dom

### PR DESCRIPTION
**Pull request by kpuputti**
- [X] include r-dom in project dependencies
 - [x] convert all components to use r-dom instead of JSX
 - [x] change all .jsx file suffixes to .js
 - [x] remove all jsx configuration (webpack, eslint, babel)

### CODE REVIEW

File: client/app/components/onboardingGuide/GuideFilterPage.jsx [hunk 1]
```diff
@@ -1,50 +1,42 @@
...
+
+    pageData.info_image ?
+      div({ className: css.sloganImageContainerBig }, [
+        img({
+          className: css.sloganImage,
+          src: pageData.info_image,
+          alt: t('info_image_alt'),
+        }),
+      ]) :
+    null,
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideFilterPage.jsx]: I guess, this should be indented more.
('if's become quite hard to follow with this style)

File: client/app/components/onboardingGuide/GuideStatusPage.jsx [hunk 2]
```diff
@@ -11,78 +13,66 @@ const GuideStatusPage = (props) => {
 
   // TODO: interpolating translation strings needs more thinking
   const title = props.nextStep ?
-    props.t('title').replace(/%\{(\w+)\}/g, props.name) :
-    props.t('title_done');
-
+          props.t('title').replace(/%\{(\w+)\}/g, props.name) :
+        props.t('title_done');
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideStatusPage.jsx]: Indetation looks strange.

File: client/app/components/onboardingGuide/GuideCoverPhotoPage.jsx [hunk 3]
```diff
@@ -1,47 +1,41 @@
...
+
+    pageData.info_image ?
+      div({ className: css.sloganImageContainer }, [
+        img({
+          className: css.sloganImage,
+          src: pageData.info_image,
+          alt: t('info_image_alt'),
+        }),
+      ]) :
+    null,
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideCoverPhotoPage.jsx]: I guess, this should be indented more.
('if's become quite hard to follow with this style)

File: client/app/components/onboardingGuide/GuideCoverPhotoPage.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    pageData.info_image ?
+      div({ className: css.sloganImageContainer }, [
+        img({
+          className: css.sloganImage,
+          src: pageData.info_image,
+          alt: t('info_image_alt'),
+        }),
+      ]) :
+    null,
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideCoverPhotoPage.js]: I guess, this should be indented more.
('if's become quite hard to follow with this style)

File: client/app/components/onboardingGuide/GuideStatusPage.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    e.preventDefault();
+    props.changePage(path);
+  };
+
+  const onboardingData = props.onboarding_data;
+
+  // TODO: interpolating translation strings needs more thinking
+  const title = props.nextStep ?
+          props.t('title').replace(/%\{(\w+)\}/g, props.name) :
+        props.t('title_done');
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideStatusPage.js]: Indetation looks strange.

File: client/app/components/onboardingGuide/GuideStatusPage.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        ]),
+      ]);
+    }))),
+
+    props.nextStep ?
+      a({
+        className: css.nextButton,
+        href: `${props.initialPath}/${props.nextStep.link}`,
+        onClick: (e) => handleClick(e, `/${props.nextStep.link}`),
+      }) :
```
Comment (Gnito) [inline client/app/components/onboardingGuide/GuideStatusPage.js]: }, [props.t('add_your_own')]):

File: client/app/components/onboardingGuide/GuideCoverPhotoPage.js (hunk 4 above)
Comment (kpuputti) [inline client/app/components/onboardingGuide/GuideCoverPhotoPage.js]: Which part do you mean? The ternary expression is at the same level as the `h2()` and `p()` above, and everything inside is already indented. Or do you mean just the `null`?
